org.eclipse.xtext/src/org/eclipse/xtext/resource/impl/IsAffectedExtension.java

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Sebastian Zarnekow and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.resource.impl;
+
+import java.util.Collection;
+import java.util.List;
+
+import org.eclipse.xtext.resource.IResourceDescription;
+import org.eclipse.xtext.resource.IResourceDescriptions;
+import org.eclipse.xtext.resource.IResourceDescription.Delta;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Binding;
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import com.google.inject.Provider;
+import com.google.inject.TypeLiteral;
+
+/**
+ * An extension that can be bound to compute resources as being affected
+ * by a change beyond the default linking semantics, e.g. some validations
```

Comment:
Maybe some rewording to improve the readability: `An extension that can be bound to compute resources beyond the default linking semantics as being affected by a change`